src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -195,9 +196,13 @@ public ListenerToken AddDocumentReplicationListener([NotNull]EventHandler<Docume
             [NotNull]EventHandler<DocumentReplicationEventArgs> handler)
         {
             CBDebug.MustNotBeNull(WriteLog.To.Sync, Tag, nameof(handler), handler);
-            Config.Options.ProgressLevel = ReplicatorProgressLevel.PerDocument;
             var cbHandler = new CouchbaseEventHandler<DocumentReplicationEventArgs>(handler, scheduler);
+            if (_repl!= null && _documentEndedUpdateEventsCount <= 0) {
```

Comment:
This is starting to break thread safety.  The `Event` class here already reports how many events are present when adding and removing.  I think it would be safer to add a method to get the current count from inside the event rather than tracking it separately like this.

Reply:
The count is not accessible in Event. I think I did try that first.